Kernel/ACPI/ACPIStaticParser.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ACPI/ACPIParser.h>
+
+class ACPIStaticParser : ACPIParser {
+public:
+    static void initialize(ACPI_RAW::RSDPDescriptor20& rsdp);
+    static void initialize_without_rsdp();
+    static bool is_initialized();
+
+    virtual ACPI_RAW::SDTHeader* find_table(const char* sig) override;
+    virtual void do_acpi_reboot() override;
+    virtual void do_acpi_shutdown() override;
+    virtual bool is_operable() override { return m_operable; }
+
+protected:
+    ACPIStaticParser();
+    explicit ACPIStaticParser(ACPI_RAW::RSDPDescriptor20&);
+
+    virtual void mmap(VirtualAddress preferred_vaddr, PhysicalAddress paddr, u32) override;
+    virtual void mmap_region(Region& region, PhysicalAddress paddr) override;
+
+private:
+    void locate_static_data();
+    void locate_all_aml_tables();
+    void locate_main_system_description_table();
+    void initialize_main_system_description_table();
+    ACPI_RAW::SDTHeader& find_fadt();
+    void init_fadt();
+    ACPI_RAW::RSDPDescriptor20* search_rsdp();
+
+    // Early pointers that are needed really for initializtion only...
+    ACPI_RAW::RSDPDescriptor20* m_rsdp;
+    ACPI_RAW::SDTHeader* m_main_system_description_table;
+
+    RefPtr<ACPI::MainSystemDescriptionTable> m_main_sdt;
+    RefPtr<ACPI::FixedACPIData> m_fadt;
+    RefPtr<ACPI::MCFG> m_mcfg;
+    RefPtr<ACPI::MADT> m_madt;
```

Comment:
I forgot to ask earlier, why are these objects reference-counted? Are they going to have many owners who share the responsibility over them, or are they all logically owned by this `ACPIStaticParser` object?
If they don't need multiple owners, they don't need to be reference-counted, but can instead use `OwnPtr` without inheriting from `RefCounted`.
I'm asking because I don't see any other `RefPtr`'s pointing to these objects.

Reply:
`m_madt` and `m_mcfg`, at least as I planned it beforehand, should be shared also with other classes that do not exist right now (MADT is the table where you have info about processors and APIC routing rules, and MCFG is the table for PCI ECAM descriptors).
What I'll do is to just delete those pointers as they are not used right now. `m_fadt` & `m_main_sdt` can be `OwnPtr`.